flamingpy/utils/viz.py

```diff
@@ -301,34 +371,83 @@ def draw_dec_graph(graph, label_edges=True, node_labels=None, title=""):
     if not isinstance(graph.graph, nx.Graph):
         raise ValueError("The graph must be implemented with the networkx backend.")
     graph = graph.graph
-    plt.figure()
+    layout = nx.circular_layout(graph)
+
+    fig, ax = plt.subplots()
     if title != "":
-        plt.title(title, family="serif", size=10)
+        ax.set_title(title)
     # NetworkX drawing function for circular embedding of graphs.
+
+    # Color edges based on weight, and draw a colobar.
+
+    # divider = make_axes_locatable(ax)
+    ax.axis("off")
+    cmap, norm = draw_curved_edges(graph, layout, ax)
+    nx.draw_networkx_nodes(graph, pos=layout, node_color="#202020", ax=ax)
+
+    # Draw node labels
     if node_labels is not None:
         node_labels = {node: label for node, label in node_labels.items() if node in graph}
-    nx.draw_circular(
-        graph,
-        edgelist=[],
-        with_labels=node_labels is not None,
-        labels=node_labels,
-        node_color="k",
-        font_size=7,
-        font_color="w",
-        font_family="serif",
-    )
-    # Color edges based on weight, and draw a colobar.
-    weight_list = [graph.edges[edge]["weight"] for edge in graph.edges]
-    weight_dict = {edge: "{:.2f}".format(graph.edges[edge]["weight"]) for edge in graph.edges}
+        nx.draw_networkx_labels(
+            graph, pos=layout, labels=node_labels, font_color="white", font_weight=100, ax=ax
+        )
+
     if label_edges:
+        weight_dict = {edge: "{:.2f}".format(graph.edges[edge]["weight"]) for edge in graph.edges}
         nx.draw_networkx_edge_labels(
-            graph, nx.circular_layout(graph), edge_labels=weight_dict, font_size=7
+            graph,
+            layout,
+            edge_labels=weight_dict,
+            font_size=plot_params.get("font.size", 7) * 0.75,
+            clip_on=False,
+            alpha=0.7,
+            verticalalignment="center_baseline",
+            bbox={"alpha": 0},
         )
-    r = nx.draw_networkx_edges(graph, nx.circular_layout(graph), edge_color=weight_list)
-    cbar = plt.colorbar(r)
-    cbar.ax.tick_params(labelsize=10)
 
+    cax, kw = mpl.colorbar.make_axes(ax, location="right", fraction=0.15)
+    cbar = mpl.colorbar.ColorbarBase(cax, cmap=cmap, norm=norm, **kw)
+    cbar.ax.tick_params(labelsize=plot_params.get("axes.labelsize", 10), rotation=270)
+    cbar.set_label(
+        "weight", rotation=270, fontsize=plot_params.get("axes.labelsize", 10), labelpad=20
+    )
+
+    axs = [ax, cax]
+    return fig, axs
 
+
+def draw_curved_edges(graph, layout, ax, rad=0.5):
```

Comment:
Is this hard to achieve & customize using nx layout functions?

Reply:
Networkx has a wrapper around matplotlib and it is quite limited. To achieve the curved edge is necessary to use the full machinery of matplotlib.